MSAL/src-internal/ios/MSALLegacySharedADALAccount.m

```diff
@@ -0,0 +1,200 @@
+        
+        if (_authority.tenant.type == MSIDAADTenantTypeCommon)
+        {
+            _identifier = [MSIDAccountIdentifier homeAccountIdentifierFromUid:_objectId utid:_tenantId];
+        }
+        
+        NSMutableDictionary *claims = [NSMutableDictionary new];
+        
+        if (![NSString msidIsStringNilOrBlank:_objectId])
+        {
+            claims[@"oid"] = _objectId;
+        }
+        
+        if (![NSString msidIsStringNilOrBlank:_tenantId])
+        {
+            claims[@"tid"] = _tenantId;
+        }
+        
+        NSString *displayName = [jsonDictionary msidStringObjectForKey:@"displayName"];
+        
+        if (![NSString msidIsStringNilOrBlank:displayName])
+        {
+            claims[@"name"] = displayName;
+        }
+        
+        _username = [jsonDictionary msidStringObjectForKey:@"username"];
+        _accountClaims = claims;
+        
+        MSID_LOG_WITH_CTX_PII(MSIDLogLevelInfo, nil, @"Created external ADAL account with identifier %@, object Id %@, tenant Id %@, name %@, username %@, claims %@", MSID_PII_LOG_TRACKABLE(_identifier), MSID_PII_LOG_MASKABLE(_objectId), _tenantId, MSID_PII_LOG_MASKABLE(displayName), MSID_PII_LOG_EMAIL(_username), MSID_PII_LOG_MASKABLE(_accountClaims));
+    }
+    
+    return self;
+}
+
+#pragma mark - Match
+
+- (BOOL)matchesParameters:(MSALAccountEnumerationParameters *)parameters
+{
+    BOOL matchResult = YES;
+    
+    if (parameters.identifier)
+    {
+        matchResult &= ([self.identifier caseInsensitiveCompare:parameters.identifier] == NSOrderedSame);
+    }
+    
+    if (parameters.username)
+    {
+        matchResult &= ([self.username caseInsensitiveCompare:parameters.username] == NSOrderedSame);
+    }
+    
+    if (parameters.tenantProfileIdentifier)
+    {
+        matchResult &= ([self.objectId caseInsensitiveCompare:parameters.tenantProfileIdentifier] == NSOrderedSame);
+    }
+    
+    return matchResult &= [super matchesParameters:parameters];
+}
+
+#pragma mark - Updates
+
+- (BOOL)updateAccountWithMSALAccount:(id<MSALAccount>)account
+                     applicationName:(NSString *)appName
+                           operation:(MSALLegacySharedAccountWriteOperation)operation
+                      accountVersion:(MSALLegacySharedAccountVersion)accountVersion
+                               error:(NSError **)error
+{
+    BOOL result = [super updateAccountWithMSALAccount:account
+                                      applicationName:appName
+                                            operation:operation
+                                       accountVersion:accountVersion
+                                                error:error];
+    
+    if (!result)
+    {
+        return NO;
+    }
+    
+    _username = account.username;
+    return YES;
+}
+
+- (NSDictionary *)updatedFieldsWithAccount:(id<MSALAccount>)account
+{
+    NSMutableDictionary *updatedFields = [NSMutableDictionary new];
```

Comment:
Can we just create a NSDictionary here instead of NSMutableDictionary as it just contains one key-value pair.

Reply:
we're likely going to add more fields in the future, but sure :)

Reply:
Addressed